### DISCUSSION TITLE
Can Rcursor use lua to find different authoritative DNS based on the request source IP

### DISCUSSION THREAD

**Discussion by sysiocn** [PowerDNS/pdns#14298] [Q&A]
![image](https://github.com/PowerDNS/pdns/assets/169319943/170fa523-6d4b-435d-b57d-63e887ab9c8b)
1，Can recursor find different authoritative servers based on the source IP or network segment to obtain different resolution information?

**Comment by omoerbeek** [comment]
No, recursor cannot do that, but authoritative servers can use EDNS Client Subnet to vary their responses based on client network info. You need to setup that in your auth(s) and also have to configure the recursor to pass ECS info to these auths. The recursor caches will take the ECS info into account when constructing answers from the cache.

Alternatively, you might wan to look into dnsdist, which can route queries based on many client or request properties.

**Reply by sysiocn** [reply]
Hello, does dnsdist need to be implemented in conjunction with lua?

**Comment by SkySai1** [comment]
You can use dnsdist instead recursor with using EDNS client subnet or Proxy. Then you need enable LUA records and ECS working on pdns authority and finally create RR with LUA type and view functions

https://doc.powerdns.com/authoritative/lua-records/
